trackintel/preprocessing/util.py

```diff
@@ -107,7 +107,11 @@ def _explode_agg(column, agg, orig_df, agg_df):
     temp = agg_df.explode(column)
     temp = temp[temp[column].notna()]
     temp.index = temp[column]
-    return orig_df.join(temp[agg], how="left")
+
+    return_df = orig_df.join(temp[agg], how="left")
```

Comment:
As dtype for index is now more flexible, the join function will create an `object` dtype index, even if the index does not change for `orig_df`. We add another line here to ensure consistency of index dtype

Reply:
good catch